pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go

```diff
@@ -88,10 +88,26 @@ func (gce *GCECloud) ensureInternalLoadBalancer(clusterName, clusterID string, s
 		}
 	}
 
+	// Determine IP which will be used for the expected forwading rule
+	// If no IP has been established in the forwarding rule or specified in the Service spec,
+	// then targetIP = "".
+	targetIP := determineTargetIP(svc, existingFwdRule)
+
+	var addrMgr *addressManager
+	if targetIP != "" {
```

Comment:
Just curious. The external LB ensures a static IP (i.e., creating a new one) even if `targetIP == ""` because many other components (e.g., firewall, targetpool) needs a target IP. I guess this not the case for ILB?

Reply:
The firewall and target pool only use the IP for providing some extra info in the description.  On first sync the description IP would be empty; but on second sync, it would be filled. I didn't see a need to perform the extra 2 API calls when the forwarding rule will just generate an IP itself, but I'll change the logic to have the description set right from the first sync.